src/components/App/ExpressRecorder.tsx

```diff
@@ -526,6 +518,8 @@ export class ExpressRecorder extends Component<ExpressRecorderProps, State> {
                     .catch(e => {
                         this.handleError("Failed to allocate resource: " + e.message);
                     });
+            } else {
+                this.setState({ stream: undefined });
```

Comment:
something is not right with approaching all Promises here.
you may run 2 `setState` calls async without handing the resolve logic properly. (line **507** and line **522**)
this is a race situation where you might lose control of resolving promises.
if you wish to update the state's stream object only, then create a dedicated private function for it.
plus, use callback functions or Promise objects to control the execution time of the React event loops.

Reply:
you right..
I reduced setState calls - 1 instead of 3